src/duplicacy_gcdstorage.go

```diff
@@ -45,11 +46,19 @@ type GCDConfig struct {
 }
 
 func (storage *GCDStorage) shouldRetry(threadIndex int, err error) (bool, error) {
+	const LIMIT_BACKOFF_TIME = 64
+	const MAX_NUMBER_OF_RETRIES = 15
+	minimumSleepRatio := 0.1
+	maximumSleepRatio := 0.2
+	minimumSleep := float64(storage.numberOfThreads) * minimumSleepRatio
+	maximumSleep := float64(storage.numberOfThreads) * maximumSleepRatio
+	rand.Seed(time.Now().UnixNano()) // unsure if this is needed
```

Comment:
It's needed once per program execution, but not once per method call, otherwise you're just wasting CPU cycled re-initializing it each time. It should be called near the top of `main()` if it's not already being called there, and then not called here.